### PR TITLE
Feature/nw23000322 add const

### PR DESCRIPTION
**Pull request by FiorenzaBusi**
## Description

Added a new feature for solving constant variables, based on the CONST01 test case.
Added more test cases to have a complete coverage of the new feature.

## Description
Modified `data_definitions.kt` to manage the definition of a constant numeric variable without using the CONST keyword.
Modified `expression.kt` to handle multi-line evaluation of an alphanumeric constant variable.
Furthermore, the modification has been prepared in order to be able to throw the error in case of definition of hexadecimal constant variable.
Created 2 test cases (`CONST01.rpgle` and `CONST02.rpgle`) which were then defined in the `InterpreterTest.kt` in order to test the changes.
Added `MUTE01_07.rpgle` containing all possible test cases for constants (except hexadecimal ones), which was then declared in `MuteExecutionTest.kt`

## Checklist:
- [X] There are tests regarding this feature
- [X] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [X] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

**Review by lanarimarco — CHANGES_REQUESTED** [review]
A little change about use of optional and elvis operator, and just one question: It's not clear to me how you have managed the constant initialization when  the literal (value inside '') is in two or more lines eparated by minus sign

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt [hunk 1]
```diff
@@ -390,11 +391,24 @@ internal fun RpgParser.DspecContext.toAst(
...
+            initializationValue = initializationValue,
+            position = this.toPosition(true))
+}
+
 internal fun RpgParser.Dcl_cContext.toAst(
     conf: ToAstConfiguration = ToAstConfiguration()
 ): DataDefinition {
     // TODO: check more examples of const declaration
-    val initializationValueExpression: Expression = this.keyword_const().simpleExpression().toAst(conf)
+    val initializationValueExpression = if (this.keyword_const() != null) this.keyword_const().simpleExpression().toAst(conf) else this.literal().toAst(conf)
```
[the remark this answers is not in the document]
Comment (FiorenzaBusi) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt]: Thanks, for suggestion.
Now I replace if-else using elvis operator.

In regards to this question: 

> "It's not clear to me how you have managed the constant initialization when the literal (value inside '') is in two or more lines eparated by minus sign"

Instantiate a literal constant variable, across multiple lines, resulted in the content being retrieved from the "children" node.

The management logics of the minus or plus sign, in the alphanumeric multi-lines, are present in `RpgLexer.g4`, below is the piece of code in question:
```
//  Any char except +,- or ", or a + or - followed by more than just whitespace 
DblStringContent: ~["\r\n]+ -> type(StringContent); 
DblStringLiteralEnd: ["] -> popMode,type(StringLiteralEnd);
```

[the remark this answers is not in the document]
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt]: Perfect

**Review by lanarimarco — APPROVED** [review]
One of the pull requests better commented I ever seen!!!

**Pull request merged**